### PR TITLE
ci: update solana and flutter actions

### PR DESCRIPTION
**Pull request by justinenerio**
## Changes

update solana and flutter docker images to actions

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [ ] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: .github/workflows/check_pr.yml [hunk 1]
```diff
@@ -48,20 +48,26 @@ jobs:
...
 
     steps:
       - uses: actions/checkout@v2
       - uses: dart-lang/setup-dart@v1.3
 
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
+          cache: true
+      - run: flutter --version
```
Comment (ookami-kb) [inline .github/workflows/check_pr.yml:63]: Not needed.

File: .github/workflows/check_pr.yml [hunk 2]
```diff
@@ -48,20 +48,26 @@ jobs:
...
-        image: solanalabs/solana:stable
-        options: --entrypoint="solana-test-validator"
 
     steps:
       - uses: actions/checkout@v2
       - uses: dart-lang/setup-dart@v1.3
 
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: 'stable'
```
Comment (ookami-kb) [inline .github/workflows/check_pr.yml:61]: Let's use specific version everywhere for reproducible builds.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: .github/workflows/check_pr.yml [hunk 3]
```diff
@@ -8,6 +8,9 @@ concurrency:
   group: ${{ github.head_ref }}
   cancel-in-progress: true
 
+env:
+  FLUTTER_VERSION: ${{ github.event.inputs.version }}
```
Comment (ookami-kb) [inline .github/workflows/check_pr.yml:12]: It's not provided anywhere.

Comment (justinenerio) [reply .github/workflows/check_pr.yml]: Updated to hardcoded version

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**